kapitan/cli.py

```diff
@@ -38,250 +39,268 @@
-
-        compile_targets(args.inventory_path, search_paths, args.output_path,
-                        args.parallelism, args.targets, ref_controller,
-                        prune=(args.prune), indent=args.indent, reveal=args.reveal,
-                        cache=args.cache, cache_paths=args.cache_paths)
-
-    elif cmd == 'inventory':
-        if args.pattern and args.target_name == '':
-            parser.error("--pattern requires --target_name")
+        ref_controller = RefController(secrets_path)
+
+        compile_targets(inventory_path, search_paths, output_path,
+                        parallelism, targets, ref_controller,
+                        prune=(prune), indent=indent, reveal=reveal,
+                        cache=cache, cache_paths=cache_paths)
+
+    def lint(self, fail_on_warning=from_dot_kapitan('lint', 'fail-on-warning', False), 
+            skip_class_checks=from_dot_kapitan('lint', 'skip-class-checks', False),
+            skip_yamllint=from_dot_kapitan('lint', 'skip-yamllint', False),
+            search_secrets=from_dot_kapitan('lint', 'search-secrets', False),
+            secrets_path=from_dot_kapitan('lint', 'secrets-path', './secrets'),
+            compiled_path=from_dot_kapitan('lint', 'compiled-path', './compiled'),
+            inventory_path=from_dot_kapitan('lint', 'inventory-path', './inventory')):
+        '''
+        linter for inventory and secrets
+
+        Args:
+            fail_on_warning: bool
+                exit with failure code if warnings exist, default is False
+            skip_class_checks: bool
+                skip checking for unused classes, default is False
+            skip_yamllint: bool
+                skip running yamllint on inventory, default is False
+            search_secrets: bool
+                searches for plaintext secrets in inventory, default is False
+            secrets_path: str
+                set secrets path, default is "./secrets"
+            compiled_path: str
+                set compiled path, default is "./compiled"
+            inventory_path: str
+                set inventory path, default is "./inventory"
+        '''
+        start_lint(fail_on_warning, skip_class_checks,skip_yamllint, inventory_path, search_secrets, secrets_path, compiled_path)
+
+    def init(self, directory=from_dot_kapitan('init', 'directory', '.')):
+        '''
+        initialize a directory with the recommended kapitan project skeleton.
+
+        Args:
+            directory: str
+                set path, in which to generate the project skeleton, assumes directory already exists. default is "./"
+        '''
+        initialise_skeleton(directory)
+
+    def secrets(self, write=None, update=None, update_targets=from_dot_kapitan('secrets', 'update-targets', False), 
+                validate_targets=from_dot_kapitan('secrets', 'validate-targets', False),
+                b64e=from_dot_kapitan('secrets', 'base64', False),
```

Comment:
Do not remove or rename existing flags as it breaks backwards compatibility

Reply:
Sorry for this. Due to conflict of name of library and flag I changed the flag. It is fixed as I have now used an alias for library while importing.